Added Reviews model/controller and unit tests

joeziemba opened the pull request:
Installed FactoryBot
Defined user, venue, review factories
Added Review migration
Added Review model with associations and validations
Added Reviews Controller (empty for now)
Added unit tests for Reviews (ratings, associations)

baconprestosh committed:
Added Reviews controller/model/migration

baconprestosh committed:
Review unit tests in progress

baconprestosh committed:
fixed merge conflicts

baconprestosh committed:
added numericality validation to Review, assosiations for Review and Users/Venues

baconprestosh committed:
added FactoryBot to Gemfile

joeziemba committed:
Added factories for user, review, venue; added to review unit tests

joeziemba committed:
Fixed 14 code climate errors

jdmacmurtrie commented on app/assets/javascripts/review.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
You don't need this file

jdmacmurtrie commented on spec/factories.rb:
```diff
@@ -0,0 +1,36 @@
+    #       )
+    #   end
+    # end
+  end
```
Clean up commented code before submitting a PR please

jdmacmurtrie commented on spec/factories.rb:
```diff
@@ -0,0 +1,36 @@
+    user
+    venue
+  end
+end
```
These factories should all be divided into separate files.

joeziemba committed:
Split factories into seperate files, added /tmp to gitignore

joeziemba commented:
@jdmacmurtrie 
Deleted review.coffee
Split factories.rb into 3 files (one for each: user/venue/review) in factories folder
Added /tmp to .gitignore
